src/cressi_goa.c

```diff
@@ -377,48 +382,102 @@ cressi_goa_device_set_fingerprint (dc_device_t *abstract, const unsigned char da
 	return DC_STATUS_SUCCESS;
 }
 
+struct cressi_goa_irda_api_conf {
+	unsigned char id_version;
+	unsigned int idlen;
+	unsigned char logbook_entry_len;
+	unsigned int logbook_fp_offset;
+	unsigned char cmd_logbook;
+	dc_buffer_t *id;
+};
+
 static dc_status_t
-cressi_goa_device_foreach (dc_device_t *abstract, dc_dive_callback_t callback, void *userdata)
+cressi_goa_read_id(dc_device_t *abstract, struct cressi_goa_irda_api_conf *conf)
 {
-	dc_status_t status = DC_STATUS_SUCCESS;
+	static const struct cressi_goa_irda_api_conf version_conf[] = {
+		{ 0,  9, SZ_HEADER, FP_OFFSET, CMD_LOGBOOK,    NULL },
+		/*   11 is the new response length to the `CMD_VERSION` command
+		 *   15 is the length of an entry in the Logbook
+		 *    3 is the offset to the START date in the Logbook header
+		 * 0x23 is the new command to request the Logbook
+		 */
+		{ 1, 11,        15,         3, CMD_LOGBOOK_V4, NULL },
```

Comment:
Once the macros like SZ_HEADER and FP_OFFSET are added to this table, the macros themselves become pretty useless because we'll always use them indirectly through the entries in the table. So remove the macros to make it clear they are not really constant anymore.

Keep the macro for the commands because for debugging logs it's very useful to have an overview of all available commands at one place.